pkg/util/kubernetes/apiserver/hpa_controller_test.go

```diff
@@ -0,0 +1,342 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/custommetrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+	"k8s.io/api/autoscaling/v2beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/hpa"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+)
+
+func newFakeConfigMapStore(t *testing.T, ns, name string, metrics []custommetrics.ExternalMetricValue) (custommetrics.Store, kubernetes.Interface) {
+	client := fake.NewSimpleClientset()
+	store, err := custommetrics.NewConfigMapStore(client, ns, name)
+	require.NoError(t, err)
+	err = store.SetExternalMetricValues(metrics)
+	require.NoError(t, err)
+	return store, client
+}
+
+func newFakeHorizontalPodAutoscaler(name, ns string, uid string, metricName string, labels map[string]string) *v2beta1.HorizontalPodAutoscaler {
+	return &v2beta1.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+			UID:       types.UID(uid),
+		},
+		Spec: v2beta1.HorizontalPodAutoscalerSpec{
+			Metrics: []v2beta1.MetricSpec{
+				{
+					Type: v2beta1.ExternalMetricSourceType,
+					External: &v2beta1.ExternalMetricSource{
+						MetricName: metricName,
+						MetricSelector: &metav1.LabelSelector{
+							MatchLabels: labels,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newFakeAutoscalerController(client kubernetes.Interface, itf LeaderElectorInterface, dcl hpa.DatadogClient) (*AutoscalersController, informers.SharedInformerFactory) {
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+
+	autoscalerController := NewAutoscalersController(
+		client,
+		itf,
+		dcl,
+		informerFactory.Autoscaling().V2beta1().HorizontalPodAutoscalers(),
+	)
+	autoscalerController.autoscalersListerSynced = alwaysReady
+
+	return autoscalerController, informerFactory
+}
+
+var alwaysLeader = fakeLeaderElector{true}
```

Comment:
Nit: can you make this a pointer `&fakeLeaderElector{true}`so you can pass it directly to 

`newFakeAutoscalerController(client, &alwaysLeader, d)`